src/lib/CheckoutWithCard.ts

```diff
@@ -1,164 +1,178 @@
-    paperDomain
-  );
-
-  const checkoutWithCardLink = new LinksManager(CheckoutWithCardUrlBase);
-  checkoutWithCardLink.addClientSecret(sdkClientSecret);
-  checkoutWithCardLink.addStylingOptions(options);
-  checkoutWithCardLink.addLocale(locale);
-  checkoutWithCardLink.addAppName(appName);
-
-  return checkoutWithCardLink.getLink();
+	const paperDomain = useAltDomain ? PAPER_APP_URL_ALT : PAPER_APP_URL;
+
+	const CheckoutWithCardUrlBase = new URL(
+		CHECKOUT_WITH_CARD_IFRAME_URL,
+		paperDomain
+	);
+
+	const checkoutWithCardLink = new LinksManager(CheckoutWithCardUrlBase);
+	checkoutWithCardLink.addClientSecret(sdkClientSecret);
+	checkoutWithCardLink.addStylingOptions(options);
+	checkoutWithCardLink.addLocale(locale);
+	checkoutWithCardLink.addAppName(appName);
+
+	return checkoutWithCardLink.getLink();
 }
 
 export interface CheckoutWithCardMessageHandlerArgs {
-  iframe: HTMLIFrameElement;
-  onPaymentSuccess?: ({ id }: { id: string }) => void;
-  onReview?: (result: ReviewResult) => void;
-  onError?: (error: PaperSDKError) => void;
-  onOpenKycModal: (props: KycModal) => void;
-  onCloseKycModal: () => void;
+	iframe: HTMLIFrameElement;
+	onPaymentSuccess?: ({ id }: { id: string }) => void;
+	onReview?: (result: ReviewResult) => void;
+	onError?: (error: PaperSDKError) => void;
+	onOpenKycModal: (props: KycModal) => void;
+	onCloseKycModal: () => void;
 }
 
 export function createCheckoutWithCardMessageHandler({
-  iframe,
-  onError,
-  onOpenKycModal,
-  onCloseKycModal,
-  onReview,
-  onPaymentSuccess,
+	iframe,
+	onError,
+	onOpenKycModal,
+	onCloseKycModal,
+	onReview,
+	onPaymentSuccess,
 }: CheckoutWithCardMessageHandlerArgs) {
-  return (event: MessageEvent) => {
-    if (!event.origin.startsWith(paperDomain)) {
-      return;
-    }
-    const { data } = event;
-    switch (data.eventType) {
-      case "checkoutWithCardError":
-        if (onError) {
-          onError({
-            code: data.code as PaperSDKErrorCode,
-            error: data.error,
-          });
-        }
-        break;
-
-      case "paymentSuccess":
-        if (onPaymentSuccess) {
-          onPaymentSuccess({ id: data.id });
-        }
-        break;
-
-      case "reviewComplete":
-        if (onReview) {
-          onReview({
-            id: data.id,
-            cardholderName: data.cardholderName,
-          });
-        }
-        break;
-
-      case "openModalWithUrl":
-        onOpenKycModal({ iframeLink: data.url });
-        break;
-
-      case "completedSDKModal":
-        onCloseKycModal();
-
-        if (data.postToIframe) {
-          postMessageToIframe(iframe, data.eventType, data);
-        }
-        break;
-      case "sizing": {
-        iframe.style.height = data.height + "px";
-        iframe.style.maxHeight = data.height + "px";
-        break;
-      }
-      default:
-      // Ignore unrecognized event
-    }
-  };
+	return (event: MessageEvent) => {
+		if (
+			!event.origin.startsWith(PAPER_APP_URL) ||
```

Comment:
Don't have a clean way to access `useAltDomain` or `paperDomain` from here

Reply:
we could pass it in as a prop or this works too actl